CIScripts/Checkout/Zuul.ps1

```diff
@@ -3,23 +3,21 @@
 function Get-ZuulRepos {
     Param (
         [Parameter(Mandatory = $true)] [string] $GerritUrl,
-        [Parameter(Mandatory = $false)] [string] $ZuulProject,
-        [Parameter(Mandatory = $false)] [string] $ZuulRef,
-        [Parameter(Mandatory = $false)] [string] $ZuulUrl,
-        [Parameter(Mandatory = $false)] [string] $ZuulBranch
+        [Parameter(Mandatory = $true)] [string] $ZuulBranch,
+        [Parameter(Mandatory = $false)] [hashtable] $ZuulAdditionalParams
     )
 
     $ZuulClonerOptions = @(
+        "--zuul-branch=$($ZuulBranch)",
         "--map=./CIScripts/clonemap.yml",
         $GerritUrl
     )
 
-    if ($ZuulUrl -ne "") {
+    if ($ZuulAdditionalParams.Count -ne 0) {
```

Comment:
Every type except basic ones (`string`, `integer`, `bool` etc.) can be null, and _will_ be null in case of non-present optional parameter. So it's best to check `if ($ZuulAdditionalParams -ne $null)` or just even `if ($ZuulAdditionalParams)`.

You could also make a class for this, so it's impossible to pass invalid hashtable, but meh.